lib/resource/resource.ex

```diff
@@ -9,24 +9,62 @@ defmodule AshPaperTrail.Resource do
     A section for configuring how versioning is derived for the resource.
     """,
     schema: [
-      ignore_attributes: [
+      attributes_as_attributes: [
         type: {:list, :atom},
         default: [],
         doc: """
-        A list of attributes that should be ignored. `created_at`, `updated_at` and the primary key are always ignored.
+        A set of attributes that should be set as attributes on the version resource, instead of stored in the freeform `changes` map attribute.
         """
       ],
-      attributes_as_attributes: [
-        type: {:list, :atom},
-        default: [],
+      belongs_to_actor: [
+        args: [:relationship_name, :resource],
         doc: """
-        A set of attributes that should be set as attributes on the version resource, instead of stored in the freeform `changes` map attribute.
+        Creates a belongs_to relationship to version resource. When creating a new version, if the actor is set and matches the 
+        resource type, the version will be related to the actor. If your actors are polymorphic or varying types, declare a 
+        belongs_to_actor for each type.
+
+        The relationship created will always be `allow_nil? true` and foreign key constraints (if any) will unassociate 
+        if the actor record is deleted. If you need more complex relationship, set `define_attribute? false` and add the
+        relationship via a mixin.
+        """,
+        schema: [
+          relationship_name: [
+            type: :atom,
+            doc: "The name of the relationship to use for the actor (e.g. :user)",
+            required: true
+          ],
+          resource: [
+            type: :module,
+            doc: "The resource of the actor (e.g. MyApp.Users.User)",
+            required: true
+          ],
+          api: [
+            type: :module,
+            doc: "The apit of the actor (e.g. MyApp.Users)",
+            required: true
+          ],
+          define_attribute?: [
+            type: :boolean,
+            default: true,
+            doc:
+              "If set to `false` an attribute is not created on the resource for this relationship, and one must be manually added in `attributes`, invalidating many other options."
+          ]
+        ]
+      ],
```

Comment:
You need to make `belongs_to_actor` into a `Spark.DSL.Entity`. Check the spark docs for more (there isn't much there but should be enough to go on). You'll also need a transformer that unwraps the list because entities are a list but your only want this specified once. It's annoying, should be a spark feature to do that automatically.

Reply:
It *should* be a nested section, but sections don't currently support positional args in spark. That is just a missing piece in spark currently.